python/ray/serve/_private/router.py

```diff
@@ -176,6 +178,9 @@ def update_deployment_config(
     ):
         """Update the config for the deployment this router sends requests to."""
 
+        if self._shutdown:
+            return
```

Comment:
are there other methods where we should add a similar check?

Reply:
This is the only place that can cause the metrics pusher to falsely restart after being shutdown (due to a delayed long poll notification from the controller)